pkg/query-service/app/logs/v3/enrich_query.go

```diff
@@ -149,3 +151,69 @@ func enrichFieldWithMetadata(field v3.AttributeKey, fields map[string]v3.Attribu
 	field.DataType = v3.AttributeKeyDataTypeString
 	return field
 }
+
+func jsonFilterEnrich(filter v3.FilterItem) v3.FilterItem {
+	// check if it is a json request
+	if !strings.HasPrefix(filter.Key.Key, "body.") {
+		return filter
+	}
+
+	// check if the value is a int, float, string, bool
+	valueType := ""
+	switch filter.Value.(type) {
+	case uint8, uint16, uint32, uint64, int, int8, int16, int32, int64:
+		valueType = "int64"
+	case float32, float64:
+		valueType = "float64"
+	case string:
+		valueType, filter.Value = parseStrValue(filter.Value.(string), filter.Operator)
+	case bool:
+		valueType = "bool"
+	}
+
+	// check if it is array
+	if strings.HasSuffix(filter.Key.Key, "[*]") {
+		valueType = fmt.Sprintf("array(%s)", valueType)
+	}
+
+	filter.Key.DataType = v3.AttributeKeyDataType(valueType)
```

Comment:
I believe only handwritten filters wouldn't have a data type, but the ones created from UI already have it. Does it affect anything if you always prefer to infer logic regardless of the type sent by the frontend?

Reply:
So if dataType of a field is not present then it will be enriched from clickhouse. 

The above logic will only run for keys having prefix `body.` and it will only happen for json query so this won't affect anything.